console_conf/controllers/identity.py

```diff
@@ -13,23 +13,56 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import json
 
 from subiquitycore.controllers.identity import BaseIdentityController
+from subiquitycore.utils import disable_first_boot_service, run_command, mark_firstboot_complete
 
 from console_conf.ui.views import IdentityView, LoginView
 
 
 class IdentityController(BaseIdentityController):
     identity_view = IdentityView
 
-    def identity(self):
+    def default(self):
         title = "Profile setup"
         excerpt = "Enter an email address from your account in the store."
         footer = ""
         self.ui.set_header(title, excerpt)
         self.ui.set_footer(footer, 40)
         self.ui.set_body(self.identity_view(self.model, self.signal, self.opts, self.loop))
 
+    def identity_done(self, email):
+        if not self.opts.dry_run:
+            self.ui.frame.body.progress.set_text("Contacting store...")
+            self.loop.draw_screen()
+            result = run_command(["snap", "create-user", "--sudoer", "--json", email])
+            self.progress.set_text("")
+            if result['status'] != 0:
+                self.ui.frame.body.error.set_text("Creating user failed:\n" + result['err'])
+                return
+            else:
+                # mark ourselves complete
+                mark_firstboot_complete()
+
+                data = json.loads(result['output'])
+                result = {
+                    'realname': email,
+                    'username': data['username'],
+                    'passwod': '',
```

Comment:
Yeah, I presume this is an irrelevant typo that can be fixed but I haven't checked. All this code is nonsense anyway.